RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -0,0 +1,965 @@
+                                    Float_t TauLeadChargedPFCandPt,
+                                    Float_t TauLeadChargedPFCandEtaAtEcalEntrance,
+                                    Float_t TauEmFraction,
+                                    Float_t TauLeadPFChargedHadrHoP,
+                                    Float_t TauLeadPFChargedHadrEoP,
+                                    Float_t TauVisMassIn,
+                                    Float_t TaudCrackEta,
+                                    Float_t TaudCrackPhi,
+                                    Float_t TauHasGsf,
+                                    Int_t TauSignalPFGammaCandsIn,
+                                    Int_t TauSignalPFGammaCandsOut,
+                                    const std::vector<Float_t>& GammasdEtaInSigCone,
+                                    const std::vector<Float_t>& GammasdPhiInSigCone,
+                                    const std::vector<Float_t>& GammasPtInSigCone,
+                                    const std::vector<Float_t>& GammasdEtaOutSigCone,
+                                    const std::vector<Float_t>& GammasdPhiOutSigCone,
+                                    const std::vector<Float_t>& GammasPtOutSigCone,
+                                    Float_t ElecEta,
+                                    Float_t ElecPhi,
+                                    Float_t ElecEtotOverPin,
+                                    Float_t ElecChi2NormGSF,
+                                    Float_t ElecChi2NormKF,
+                                    Float_t ElecGSFNumHits,
+                                    Float_t ElecKFNumHits,
+                                    Float_t ElecGSFTrackResol,
+                                    Float_t ElecGSFTracklnPt,
+                                    Float_t ElecPin,
+                                    Float_t ElecPout,
+                                    Float_t ElecEecal,
+                                    Float_t ElecDeltaEta,
+                                    Float_t ElecDeltaPhi,
+                                    Float_t ElecMvaInSigmaEtaEta,
+                                    Float_t ElecMvaInHadEnergy,
+                                    Float_t ElecMvaInDeltaEta)
+{ 
+  double sumPt  = 0.;
+  double dEta   = 0.;
+  double dEta2  = 0.;
+  double dPhi   = 0.;
+  double dPhi2  = 0.;
+  double sumPt2 = 0.;
+  for ( unsigned int i = 0 ; i < GammasPtInSigCone.size() ; ++i ) {
+    double pt_i  = GammasPtInSigCone[i];
+    double phi_i = GammasdPhiInSigCone[i];
+    if ( GammasdPhiInSigCone[i] > M_PI ) phi_i = GammasdPhiInSigCone[i] - 2*M_PI;
+    else if ( GammasdPhiInSigCone[i] < -M_PI ) phi_i = GammasdPhiInSigCone[i] + 2*M_PI;
+    double eta_i = GammasdEtaInSigCone[i];
+    sumPt  +=  pt_i;
+    sumPt2 += (pt_i*pt_i);
+    dEta   += (pt_i*eta_i);
+    dEta2  += (pt_i*eta_i*eta_i);
+    dPhi   += (pt_i*phi_i);
+    dPhi2  += (pt_i*phi_i*phi_i);
+  }
+
+  Float_t TauGammaEnFracIn = sumPt/TauPt;
```

Comment:
Please check TauPt > 0 before dividing, just like you do for sumPt in the next line.